kube-core/src/schema.rs

```diff
@@ -36,6 +37,110 @@ impl Visitor for StructuralSchemaRewriter {
                 .object
                 .get_or_insert_with(|| Box::new(ObjectValidation::default()));
             for variant in one_of {
+                // If we provide descriptions for enum variants it will produce invalid CRDs like in the following example
```

Comment:
I agree that it's good to have inline comments on what's happening (and that the comments on the existing transform are way too sparse), but I think dumping the full JSON is going a bit far in the other direction. How about something like "Move enum variant documentation from `oneOf` clause to its corresponding `property`"?